marconi/test/Integration.hs

```diff
@@ -0,0 +1,355 @@
+          (C.TxOutValue C.MultiAssetInAlonzoEra $ C.lovelaceToValue amountPaid)
+          (C.TxOutDatumHash C.ScriptDataInAlonzoEra scriptDatumHash)
+      txOut2 :: C.TxOut ctx C.AlonzoEra
+      txOut2 =
+        C.TxOut
+          (C.AddressInEra (C.ShelleyAddressInEra C.ShelleyBasedEraAlonzo) address)
+          (C.TxOutValue C.MultiAssetInAlonzoEra $ C.lovelaceToValue amountReturned)
+          C.TxOutDatumNone
+      txBodyContent :: C.TxBodyContent C.BuildTx C.AlonzoEra
+      txBodyContent =
+        C.TxBodyContent {
+          C.txIns              = [(tx1in, C.BuildTxWith $ C.KeyWitness C.KeyWitnessForSpending)],
+          C.txInsCollateral    = C.TxInsCollateralNone,
+          C.txOuts             = [txOut1, txOut2],
+          C.txFee              = C.TxFeeExplicit C.TxFeesExplicitInAlonzoEra tx1fee,
+          C.txValidityRange    = (C.TxValidityNoLowerBound, C.TxValidityNoUpperBound C.ValidityNoUpperBoundInAlonzoEra),
+          C.txMetadata         = C.TxMetadataNone,
+          C.txAuxScripts       = C.TxAuxScriptsNone,
+          C.txExtraKeyWits     = C.TxExtraKeyWitnessesNone,
+          C.txProtocolParams   = C.BuildTxWith $ Just pparams,
+          C.txWithdrawals      = C.TxWithdrawalsNone,
+          C.txCertificates     = C.TxCertificatesNone,
+          C.txUpdateProposal   = C.TxUpdateProposalNone,
+          C.txMintValue        = C.TxMintNone,
+          C.txScriptValidity   = C.TxScriptValidityNone
+        }
+  tx1body :: C.TxBody C.AlonzoEra <- H.leftFail $ C.makeTransactionBody txBodyContent
+  let
+    kw :: C.KeyWitness C.AlonzoEra
+    kw = C.makeShelleyKeyWitness tx1body (C.WitnessPaymentKey $ C.castSigningKey genesisSKey)
+    tx1 = C.makeSignedTransaction [kw] tx1body
+
+  submitTx localNodeConnectInfo tx1
+
+  -- Second transaction: spend the UTXO specified in the first transaction
+
+  _ <- liftIO $ IO.readChan indexedTxs -- wait for the first transaction to be accepted
+
+  tx2collateralTxIn <- headM . Map.keys . C.unUTxO =<< findUTxOByAddress localNodeConnectInfo (C.toAddressAny address)
+
+  (scriptTxIn, C.TxOut _ valueAtScript _) <- do
+    scriptUtxo <- findUTxOByAddress localNodeConnectInfo $ C.toAddressAny plutusScriptAddr
+    headM $ Map.toList $ C.unUTxO scriptUtxo
+
+  let lovelaceAtScript = C.txOutValueToLovelace valueAtScript
+  assert $ lovelaceAtScript == 10_000_000 -- script has the 10 ADA we put there in tx1
+
+  redeemer <- H.forAll CGen.genScriptData -- The script always returns true so any redeemer will do
+  let
+      -- The following execution unit and fee values were found by
+      -- trial and error. When the transaction which we are in the
+      -- process of creating, fails, then it will report the values it
+      -- wants. And although they change again after you correct them,
+      -- then the procedure converges quickly.
+      executionUnits = C.ExecutionUnits {C.executionSteps = 500_000, C.executionMemory = 10_000 }
+      tx2fee = 1000303 :: C.Lovelace
```

Comment:
Is there not a way to have this calculated? it could vary with different protocol params or how it's calculated in other eras.